pkg/controllers/cloud/networkpolicy.go

```diff
@@ -959,6 +1043,12 @@ func (a *appliedToSecurityGroup) notify(op securityGroupOperation, status error,
 		if a.hasRules {
 			return a.updateRules(r)
 		}
+	case securityGroupOperationDelete:
```

Comment:
Why is this required??

Reply:
If AT changed but np doesn't, either with some corner case or event received out of order from antrea (delete NP then delete AT), the controller never notify addrGroups since AT delete doesn't call updateRules.